### PR TITLE
ASP.NET Core DI (WIP)

### PR DESCRIPTION
**Pull request by JanOuborny**
This is the first version of the default DI implementation for ASP.NET Core.

**Current known issues:**
Using project reference instead of NuGet package
Missing unit tests

### CODE REVIEW

File: RiotSharp.AspNetCore/IServiceCollectionExtension.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            {
+                var rateLimitedRequester = new RateLimitedRequester(riotSharpOptions.TournamentApiKey.Key, 
+                    riotSharpOptions.TournamentApiKey.RateLimitPer10S, riotSharpOptions.TournamentApiKey.RateLimitPer10M);
+                serviceCollection.AddSingleton<ITournamentRiotApi>(serviceProvider => new TournamentRiotApi(rateLimitedRequester));
+            }
+            
+            return serviceCollection;
+        }
+    }
+}
```
Comment (BenFradet) [inline RiotSharp.AspNetCore/IServiceCollectionExtension.cs]: this need formatting

Comment (JanOuborny) [inline RiotSharp.AspNetCore/IServiceCollectionExtension.cs]: do you mean the empty line above `return serviceCollection;`?

Comment (BenFradet) [inline RiotSharp.AspNetCore/IServiceCollectionExtension.cs]: I just meant that there are a bunch of lines that are too long in this file.

Comment (JanOuborny) [inline RiotSharp.AspNetCore/IServiceCollectionExtension.cs]: Fixed it

**Comment by BenFradet** [comment]
@JanOuborny Do we want to fix the build before merging though?

**Comment by JanOuborny** [comment]
What build do you mean? It builds just fine for me. Do you mean the project references?

**Comment by BenFradet** [comment]
I mean appveyor not running the tests

**Comment by JanOuborny** [comment]
Yes, would be nice to have it finally clean. The build only fails because of an invalid key, so it should be easy to fix.

**Comment by BenFradet** [comment]
What do you mean?

This is the last build:

>c:\projects\RiotSharp\RiotSharp.AspNetCore\RiotSharp.AspNetCore.csproj(1,1): error MSB4041: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format

**Comment by JanOuborny** [comment]
Ups, I meant the previous builds

**Comment by JanOuborny** [comment]
Builds now. I'm gonna rebase and we are done here. I'm also gonna open an issue to change the project dependency to an NuGet one after we released, which should be resolved when we release 3.0.0

**Comment by JanOuborny** [comment]
Closed by accident

**Pull request merged**